support/globalconfig/ingress.go

```diff
@@ -18,18 +18,22 @@ func IngressConfig() *configv1.Ingress {
 }
 
 func ReconcileIngressConfig(cfg *configv1.Ingress, hcp *hyperv1.HostedControlPlane) {
-	cfg.Spec.Domain = IngressDomain(hcp)
 	if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.Ingress != nil {
 		cfg.Spec = *hcp.Spec.Configuration.Ingress
 	}
+	if cfg.Spec.Domain == "" {
+		cfg.Spec.Domain = IngressDomain(hcp)
+	}
 }
 
 func IngressDomain(hcp *hyperv1.HostedControlPlane) string {
 	if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.Ingress != nil {
 		if len(hcp.Spec.Configuration.Ingress.AppsDomain) > 0 {
 			return hcp.Spec.Configuration.Ingress.AppsDomain
 		}
-		return hcp.Spec.Configuration.Ingress.Domain
+		if len(hcp.Spec.Configuration.Ingress.Domain) > 0 {
```

Comment:
This seems like an unrelated change where if `hcp.Spec.Configuration.Ingress != nil` but  `hcp.Spec.Configuration.Ingress.Domain` is not set, we default to the `apps.$baseDomain` rather than returning `""`.

Reply:
It's related in that if you only specify only the NLB setting in config, we end up with a blank domain, unless you also specify it.